zcash_proofs/src/circuit/sapling.rs

```diff
@@ -70,22 +70,38 @@ pub struct Output<'a, E: JubjubEngine> {
 fn expose_value_commitment<E, CS>(
     mut cs: CS,
     value_commitment: Option<ValueCommitment<E>>,
-    params: &E::Params,
-) -> Result<Vec<boolean::Boolean>, SynthesisError>
-where
-    E: JubjubEngine,
-    CS: ConstraintSystem<E>,
+    params: &E::Params
+) -> Result<(Vec<boolean::Boolean>, Vec<boolean::Boolean>), SynthesisError>
+    where E: JubjubEngine,
+          CS: ConstraintSystem<E>
 {
+    // Witness the asset type
```

Comment:
And specifically compute the value commitment generator, right?

Reply:
yes, so, this is the crux of the changes, the value commitment generator now is part of the note, which is then used elsewhere. However, I'm not sure if the group hash is verified anywhere in the circuit, which is the part I'm currently trying to chase down (understanding why or why not it might be important)

Reply:
last clarification on the oops...the note contains the arbitrary u32 id of the asset type, the witness contains the value base/generator